src/data/polynomial/laurent.lean

```diff
@@ -246,6 +246,51 @@ show T n * _ = _, by
 lemma T_mul (n : ℤ) (f : R[T;T⁻¹]) : T n * f = f * T n :=
 (commute_T n f).eq
 
+/--  `trunc : R[T;T⁻¹] →+ R[X]` maps a Laurent polynomial `f` to the polynomial whose terms of
+nonnegative degree coincide with the ones of `f`.  The terms of negative degree of `f` "vanish".
+`trunc` is a left-inverse to `polynomial.to_laurent`. -/
+def trunc : R[T;T⁻¹] →+ R[X] :=
+((to_finsupp_iso R).symm.to_add_monoid_hom).comp $
+  comap_domain.add_monoid_hom $ λ a b, int.of_nat.inj
+
+@[simp]
+lemma trunc_C_mul_T (n : ℤ) (r : R) : trunc (C r * T n) = ite (0 ≤ n) (monomial n.to_nat r) 0 :=
+begin
+  apply (to_finsupp_iso R).injective,
+  rw [← single_eq_C_mul_T, trunc, add_monoid_hom.coe_comp, function.comp_app,
+    comap_domain.add_monoid_hom_apply, to_finsupp_iso_apply],
+  by_cases n0 : 0 ≤ n,
+  { lift n to ℕ using n0,
+    erw [comap_domain_single, to_finsupp_iso_symm_apply],
+    simp only [int.coe_nat_nonneg, int.to_nat_coe_nat, if_true, to_finsupp_iso_apply,
+      to_finsupp_monomial] },
+  { lift (- n) to ℕ using (neg_pos.mpr (not_le.mp n0)).le with m,
+    rw [to_finsupp_iso_apply, to_finsupp_inj, if_neg n0],
+    erw to_finsupp_iso_symm_apply,
+    ext a,
+    have := ((not_le.mp n0).trans_le (int.coe_zero_le a)).ne',
+    simp only [coeff, comap_domain_apply, int.of_nat_eq_coe, coeff_zero, single_apply_eq_zero, this,
+      forall_false_left] }
+end
+
+@[simp]
+lemma _root_.polynomial.trunc_to_laurent (f : R[X]) : trunc f.to_laurent = f :=
```

Comment:
Can you state this using [function.left_inverse](https://leanprover-community.github.io/mathlib_docs/init/function.html#function.left_inverse)?

Reply:
Riccardo, I rephrased the lemma using `left_inverse`, but it felt right to keep the applied version as well.  Let me know if you prefer me to remove the applied version.

Reply:
Having both version is totally fine.